zilliqa/src/exec.rs

```diff
@@ -915,7 +916,19 @@ impl State {
                     let error_str =
                         format!("Estimate gas failed with error: {:?}", result.exit_reason);
                     warn!(error_str);
-                    return Err(anyhow!(error_str));
+
+                    let decoded_revert_msg = extract_revert_msg(&result.return_value);
+
+                    // See: https://github.com/ethereum/go-ethereum/blob/9b9a1b677d894db951dc4714ea1a46a2e7b74ffc/internal/ethapi/api.go#L1026
+                    const REVERT_ERROR_CODE: i32 = 3;
+
+                    let response = jsonrpsee::types::ErrorObjectOwned::owned(
```

Comment:
As a minor point to consider for the future - I don't like that we're adding API-specific concerns at this level. Ideally this function would be returning an "I reverted" error and we map that to the error code at the API-layer. I'd say a similar thing for the special "Txn Hash not present" error which we have to later suppress since its not a 'real' error.

We can fix it in the future though :)